test/FileLockTestWorker.java

```diff
@@ -0,0 +1,52 @@
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.channels.FileChannel;
+import java.nio.file.StandardOpenOption;
+import java.util.Scanner;
+import java.nio.channels.FileLock;
+import java.nio.channels.OverlappingFileLockException;
+
+public class FileLockTestWorker {
+    private static String filePath = "/mnt/leofs/testlock";
+
+    private static boolean doLock(FileChannel fc, long offset, long end) throws IOException{
+        try {
+            FileLock lock = fc.tryLock(offset, end, false);
+            return lock != null;
+        } catch (OverlappingFileLockException e) {
+            return true;
+        }
+    }
+
+    public static void main(String[] args) throws IOException{
+
+        if (args.length >= 1) {
+            filePath = args[0];
+        }
+
+//        System.out.format("File: %s", filePath);
+
+        Path path = Paths.get(filePath);
+        FileChannel fc = FileChannel.open(path, StandardOpenOption.CREATE, StandardOpenOption.WRITE);
+
+        Scanner scan = new Scanner(System.in);
+
+        long offset;
+        long end;
+        boolean ret;
+
```

Comment:
I'd rather prefer to place both the generator and the parser logic in the same module for readability even if this is a tiny test program.